activerecord/lib/active_record/database_configurations/builder/config.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  class DatabaseConfigurations
+    class Builder
+      class Config # :nodoc:
```

Comment:
I think we might consider subclassing `BasicObject`.  If we `instance_eval` against a subclass object, then it exposes more methods on the DSL than we might want exposed.